tests/api2/test_apps.py

```diff
@@ -0,0 +1,288 @@
+                'ACTUAL_HOSTNAME': '0.0.0.0',
+                'ACTUAL_PORT': '31012',
+                'ACTUAL_SERVER_FILES': '/data/server-files',
+                'ACTUAL_USER_FILES': '/data/user-files',
+                'GID': '568',
+                'GROUP_ID': '568',
+                'NODE_ENV': 'production',
+                'PGID': '568',
+                'PUID': '568',
+                'TZ': 'Etc/UTC',
+                'UID': '568',
+                'USER_ID': '568'
+            },
+            'ports': [
+                {
+                    'host_ip': '0.0.0.0',
+                    'mode': 'ingress',
+                    'protocol': 'tcp',
+                    'published': 31012,
+                    'target': 31012
+                }
+            ]
+        },
+        'permissions': {
+            'command': [
+                '''
+                function process_dir() {
+                    local dir=$$1
+                    local mode=$$2
+                    local uid=$$3
+                    local gid=$$4
+                    local chmod=$$5
+                    local is_temporary=$$6
+                    # Process directory logic here...
+                }
+                process_dir /mnt/actual_budget/config check 568 568 false false
+                '''
+            ],
+            'deploy': {
+                'resources': {
+                    'limits': {
+                        'cpus': '1.0',
+                        'memory': '512m'
+                    }
+                }
+            },
+            'entrypoint': ['bash', '-c'],
+            'image': 'bash',
+            'user': 'root'
+        }
+    },
+    'x-portals': [
+        {
+            'host': '0.0.0.0',
+            'name': 'Web UI',
+            'path': '/',
+            'port': 31012,
+            'scheme': 'http'
+        }
+    ],
+    'x-notes': '''# Welcome to TrueNAS SCALE
+
+    Thank you for installing Actual Budget!
+
+    ## Documentation
+    Documentation for Actual Budget can be found at https://www.truenas.com/docs.
+
+    ## Bug reports
+    If you find a bug in this app, please file an issue at
+    https://ixsystems.atlassian.net or https://github.com/truenas/apps.
+
+    ## Feature requests or improvements
+    If you find a feature request for this app, please file an issue at
+    https://ixsystems.atlassian.net or https://github.com/truenas/apps.
+    '''
+}
+
+INVALID_YAML = '''
+services:
+  actual_budget
+    user: 568:568
+    image: actualbudget/actual-server:24.10.1
+    restart: unless-stopped
+    deploy:
+      resources: {'limits': {'cpus': '2', 'memory': '4096M'}}
+    devices: []
+    depends_on:
+      permissions:
+        condition: service_completed_successfully
+    cap_drop: ['ALL']
+    security_opt: ['no-new-privileges']
+'''
+
+
+@pytest.fixture(scope='session')
+def docker_pool():
+    with another_pool() as pool:
+        yield pool['name']
+
+
+@pytest.mark.dependency(name='docker_setup')
```

Comment:
Should this be a fixture too? Then you won't need to have `depends=['docker_setup']` in all other tests. Also you won't need to have a hack-ish last test in this file, the fixture will teardown itself correctly.